File: problems_list/剑指OfferII119.最长连续序列/剑指OfferII119-最长连续序列.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    var st=new Set();
    for(let num of nums){
        st.add(num);
    }

    let ans=0, curLen=0;
    for(let num of st){
        if(!st.has(num-1)){
            curLen=1;
            let curNum=num;
            curLen=1;
            while(st.has(curNum+1)){
                curNum=curNum+1;
                curLen++;
            }
            ans=Math.max(ans, curLen);
        }
    }
    return ans;
};

File: problems_list/剑指OfferII119.最长连续序列/剑指OfferII119-最长连续序列.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        sort(nums.begin(), nums.end());

        int ans=0, n=nums.size();
        for(int l=0, r=0; r<n; r++){
            if(r!=l && (nums[r]-nums[r-1]!=1 && nums[r]-nums[r-1]!=0)) l=r;
            ans=max(ans, nums[r]-nums[l]+1);
        }
        return ans;
    }
};